### DISCUSSION TITLE
A question about orders of processing  for a qmd document

### DISCUSSION THREAD

**Discussion by kwangkim** [quarto-dev/quarto-cli#3532] [Q&A] [answered]
I would like to know how quarto converts a qmd document.

For example, if I convert a qmd document with python and ojs code into a html with a pandoc filter.
I guess quarto runs like this.
```
qmd  ==> ipynb  ===run python ==> md ==run pandoc filter==> md ==run ojs==> html
```
Is it a correct way to process?

I also want to know whether my guess is correct for a pdf file.

```
qmd  ==> ipynb  ===run python ==> md ==run pandoc filter==> md ==run pdfxelatex==> pdf
```
Thank you.

**Comment by rgaiacs** [comment, marked as answer]
@kwangkim Thanks for the question. The conversion steps will change based on the settings.

For your executable Jupyter Notebook Python with OJS, my understanding is

1. Convert `.qmd` to `.ipynb`
2. Execute cells in `.ipynb` and store the output in `.ipynb` (done by `nbconvert`)
3. Convert `.ipynb` with code output to `.qmd`
4. Convert `.qmd` to `.html` (done with Pandoc and filters)
5. Run OJS (done in the client web browser)

For PDF, the process is similar

1. Convert `.qmd` to `.ipynb`
2. Execute cells in `.ipynb` and store the output in `.ipynb` (done by `nbconvert`)
3. Convert `.ipynb` with code output to `.qmd`
4. Convert `.qmd` to `.tex` (done with Pandoc and filters)
5. Convert `.tex` to `.pdf` (done with LaTeX engine)

Please note that OJS is **NOT** supported when rendering your document to PDF.

(small edits by @cscheid for clarification, though the original answer was 99.9% right!)

**Reply by kwangkim** [reply]
Thank you very much for your explanation.
